xdsl/transforms/riscv_register_allocation.py

```diff
@@ -1,19 +1,85 @@
+from abc import ABC
+from dataclasses import dataclass
+from typing import OrderedDict
 from xdsl.dialects.builtin import ModuleOp
 from xdsl.dialects.riscv import Register, RegisterType, RISCVOp
 from xdsl.ir import MLContext
 from xdsl.passes import ModulePass
 
 
-class RegisterAllocator:
+class AbstractRegisterAllocator(ABC):
+    """
+    Base class for register allocation strategies.
+    """
+
+    def __init__(self) -> None:
+        pass
+
+    def allocate_registers(self, module: ModuleOp) -> None:
+        """
+        Allocates unallocated registers in the module.
+        """
+
+        raise NotImplementedError()
+
+
+class RegisterAllocatorBlockNaive(AbstractRegisterAllocator):
     idx: int
 
     def __init__(self) -> None:
         self.idx = 0
 
+        """
+        Since we've got neither right now a handling of a consistent ABI nor of a calling convention,
+        let's just assume that we have all the registers available for our use except the one explicitly reserved by the default riscv ABI.
+        """
+
+        self.available_registers = OrderedDict(Register.ABI_INDEX_BY_NAME)
```

Comment:
Looking at it more in detail, it seems like I gave a bad recommendation here. Only `reserved_registers` should be a set, the others were better the way you first wrote them, as lists.

Reply:
My b